cocotb/utils.py

```diff
@@ -74,6 +69,13 @@ def get_sim_time(units=None):
 
     return result
 
+def get_precision():
+    global _LOG_SIM_PRECISION
+
+    if _LOG_SIM_PRECISION is None:
+        _LOG_SIM_PRECISION = cocotb.simulator.get_precision()  # request once and cache
+    return _LOG_SIM_PRECISION
```

Comment:
This change seems a generally useful improvement, how about doing a separate PR for this right away to keep this patch stack small?